app.js

```diff
@@ -0,0 +1,462 @@
+                tr.appendChild(td);
+                td.style.border = '1px solid black'
+                td.innerHTML = ''
+                td.style.width = '55px'
+                td.style.height = '55px'
+
+                let btn = document.createElement('button') // for each cell was created a button
+                document.body.appendChild(btn)
+                btn.style.width = '60px'
+                btn.style.height = '60px'
+                btn.style.marginLeft = '-4.5em'
+
+
+                let btnStrId = this.btnId.toString() //giving an id to each button
+                btn.id = btnStrId
+                this.buttonId.push(btn.id)
+                let col = document.getElementById(btn.id)
+                col.style.background = "grey"
+
+                let tdId = this.btnId + 'td' // giving an id to each element of the row
+                td.id = tdId
+                this.tableId.push(td.id)
+                tr.appendChild(btn)
+                this.btnId++
+
+
+            }
+            this.table.appendChild(tr);
+        }
+    }
+
+}
+function changeX(){        // changing X and Y fields at the same time (if changed separately there will be issues with opening empty fields)
+    let xValue = document.getElementById('XInput').value
+    document.getElementById('YInput').value = xValue
+}
+function changeY(){
+    let yValue = document.getElementById('YInput').value
+    document.getElementById('XInput').value = yValue
+}
+function valueSave(){ // saving the previous set x and y
+    sessionStorage.setItem('xValue',document.getElementById('XInput').value)
+    sessionStorage.setItem('yValue',document.getElementById('YInput').value)
+}
+function valueLoad(){
+    if (sessionStorage.getItem('xValue')!==null){
+        document.getElementById('XInput').value = sessionStorage.getItem('xValue')
+        document.getElementById('YInput').value = sessionStorage.getItem('yValue')
+    }
+}
+let xValue = document.getElementById('XInput').value
+let yValue = document.getElementById('YInput').value
+xValue = Number(xValue)
+yValue = Number(yValue)
+board1 = new board(xValue,yValue)
+
+function random() {
+    let randomNum = []
+    let randomMinesTemp = []
+    for (let i = 0;i<board1.size;i++){
+        randomNum.push(i)
+    }
+    for (randomNum, i = randomNum.length; i--; ) {
+        let random = randomNum.splice(Math.floor(Math.random() * (i + 1)), 1)[0];
+        randomMinesTemp.push(random) // making a new array with digits from previous array but in random order
+    }
+return randomMinesTemp
+}
+let minesAmount = (board1.size)/8
+minesAmount = Math.floor(minesAmount)
+document.getElementById("minesCounter").value = minesAmount
+let infoSize = 60*board1.x+60+board1.x*3.2+'px'     // formula for measuring size of the info menu (the one with the counter and stuff)
+document.getElementById("info").style.width = infoSize
+class field {
+    emptyTabCopy = []
+    randomMinesTemp = []
+    mines = []
```

Comment:
Field may contain single mine not list of mines. Look at classes like on lego blocks. You build an application from them. If you take car for example then car is build from doors, wheels, engine and etc. all those should be a separate class. This way you design application. Below code has to be adjusted to that.